.buildkite/pipelines/build_linux.json.py

```diff
@@ -93,20 +93,25 @@ def main(args):
             ],
         })
 
-    if os.environ.get("BUILDKITE_PULL_REQUEST", "false") != "false":
-        # We also always cross compile for aarch64 with full debug and assertions
+    # Never cross-compile for linux-aarch64 in the nightly debug build.
+    if os.environ.get("BUILDKITE_PIPELINE_SLUG", "ml-cpp") != "ml-cpp-debug-build" and \
+            os.environ.get("BUILDKITE_PULL_REQUEST", "false") != "false":
+        # Always cross compile for aarch64 with full debug and assertions
         # enabled for PR builds only. This is to detect any compilation errors
         # as early as possible.
         pipeline_steps.append({
             "label": "Build :cpp: for linux_aarch64_cross-RelWithDebInfo :linux:",
-            "timeout_in_minutes": "120",
+            "timeout_in_minutes": "150",
             "agents": {
               "cpu": "6",
               "ephemeralStorage": "20G",
               "memory": "64G",
               "image": "docker.elastic.co/ml-dev/ml-linux-aarch64-cross-build:10"
             },
             "commands": [
+              f'if [[ "{args.snapshot}" != "None" ]]; then export BUILD_SNAPSHOT={args.snapshot}; fi',
+              f'if [[ "{args.version_qualifier}" != "None" ]]; then export VERSION_QUALIFIER={args.version_qualifier}; fi',
+              "env",
```

Comment:
Is this just temporary? Printing the whole environment is a recipe for leaking secrets, so this shouldn't stay here long term.

Reply:
> 

Yes, these can now be removed.